presto-main/src/main/java/com/facebook/presto/execution/warnings/DefaultWarningCollector.java

```diff
@@ -43,9 +45,18 @@ public DefaultWarningCollector(WarningCollectorConfig config)
     public synchronized void add(PrestoWarning warning)
     {
         requireNonNull(warning, "warning is null");
-        if (warnings.size() < config.getMaxWarnings()) {
-            warnings.putIfAbsent(warning.getWarningCode(), warning);
+        switch(config.getWarningHandlingLevel()) {
+            case SUPPRESS:
+                break;
+            case NORMAL:
+                if (warnings.size() < config.getMaxWarnings()) {
+                    warnings.putIfAbsent(warning.getWarningCode(), warning);
+                }
+                break;
+            case AS_ERROR:
+                throw new PrestoException(StandardErrorCode.WARNING_AS_ERROR, warning.toString());
```

Comment:
Hmm, here you want to wrap the warning in the PrestoException so you will get better context.

Reply:
In this case, after speaking with @kaikalur, we agreed that as of now, stack traces are not useful, but we may add the stack trace context in the future for future warnings.